test/es-module/test-esm-loader-hooks.mjs

```diff
@@ -747,6 +747,29 @@ describe('Loader hooks', { concurrency: true }, () => {
     assert.strictEqual(signal, null);
   });
 
+  it('should support source maps in commonjs translator', async () => {
+    const readFile = async () => {};
+    const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+      '--no-warnings',
+      '--enable-source-maps',
+      '--experimental-loader',
```

Comment:
We might remove `--experimental-loader` in the future. Can we write this test with `--import` and `register` instead? You can pass a data URL to both `--import` and `register`, so it’s still possible to keep all the fixtures within this file. You can see an example at https://github.com/nodejs/node/blob/fba2c9332f4bccc2b0b300de234d413ae28b4b9c/test/es-module/test-esm-import-assertion-warning.mjs.

Reply:
Also followed precedent here. Can do this in a separate PR.

Reply:
This doesn't feel like a separate PR concern. The precedent is just because most of the loader tests are old, before the design change.

Reply:
Changing `--loader` to `--import` seems like a refactor that addresses a separate issue related to outdated tests.

I'm willing to handle it, but I'd prefer to approach it comprehensively and update all tests using `--loader`. This would go beyond the specific problem that this pull request is addressing.

Reply:
If you’d like to do a follow-up to update all the old tests to use the new style, that would be great. Lately we’ve been focusing on using the new style for _new_ tests, to at least not dig our hole any deeper.

Reply:
Updated!